dev-db/clickhouse/clickhouse-9999.ebuild

```diff
@@ -0,0 +1,724 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils llvm systemd toolchain-funcs user
+
+DESCRIPTION="An OSS column-oriented database management system for real-time data analysis"
+HOMEPAGE="https://clickhouse.yandex"
+LICENSE="Apache-2.0"
+
+MY_PN="ClickHouse"
+IUSE="
+	cpu_flags_x86_sse4_1 cpu_flags_x86_sse4_2 cpu_flags_x86_ssse3 cpu_flags_x86_avx
+	cpu_flags_x86_avx2 cpu_flags_x86_popcnt
+	+base64 brotli capnp +client debug doc hdfs3 hyperscan icu +jemalloc json
+	kafka libcxx libedit libressl libunwind llvm lto mongodb mysql odbc parquet
+	protobuf readline +server split-binary ssl static static-libs systemd tcmalloc test tools
+"
+
+REQUIRED_USE="
+	split-binary? ( !static-libs )
+	server? ( cpu_flags_x86_sse4_2 )
+	hdfs3? ( protobuf )
+	static? ( client server tools )
+	libressl? ( ssl )
+	readline? ( client !libedit )
+	libedit? ( client !readline )
+	libcxx? ( !llvm )
+	kafka? ( ssl !arm !x86 !elibc_Darwin !elibc_FreeBSD )
+	?? ( tcmalloc jemalloc )
+	llvm? ( !elibc_Darwin )
```

Comment:
That's not how you do it. Instead, you mask the flags in profiles for specific systems.

Reply:
I thought the rule is if the code have checks against architecture and/or glibc, then it should be reflected here, e.g. it is guaranteed to fail. My understanding of profile.mask was that it disables things by default, but people can try to workaround the problem, unmask use flag and force use flag to be used.
Here is the former case

Reply:
This REQUIRED_USE is going to mean emerge is going to tell people to disable libc which is not really what they can do.

Reply:
Are you sure? I tried that and got:
```
  The following REQUIRED_USE flag constraints are unsatisfied:
    parquet? ( !elibc_FreeBSD )

  The above constraints are a subset of the following complete expression:
```

which looks pretty reasonable

Reply:
You read that as 'if you want parquet, disable elibc_FreeBSD'. The old version of CI would flag that as impossible constraint; the check is temporarily gone due to upgrade.

Reply:
@mgorny I've pushed the update, could you please verify if I understood you correctly?